server/users/users.service.ts

```diff
@@ -69,21 +122,40 @@ class UserService {
 		return await this.getById(userId)
 	}
 
-	public async deleteUser(userId: number) {
-		const deleted = await Users.destroy({ where: { id: userId } });
+	public async updateUserSetting(userId: number, settingData: UpdateSettingCommand): Promise<UserModel> {
+		if (isEmptyObject(settingData)) {
+			throw new HttpException(400, "Incorrect input data");
+		}
 
-		if (!deleted) {
-			throw new HttpException(409, "User not found");
+		let settingUpdate = { ...{}, [settingData.setting]: (settingData.value == "true") };
```

Comment:
This is cool, but some attacker might send any value as `settingData.setting` and modify other columns as well (although only boolean ones due to `settingData.value == "true"` mapping. I'd rather we update field by field explicitly.

Reply:
Will do, thanks. 
FYI https://github.com/mbrizic/ratethegame/pull/28/commits/0713f38e2625f1f98f02c12982914cdc2f903b6e is not finished, just an intermediate step towards updating fields explicitly

Reply:
Hm, what do you mean by "update field by field explicitly"?
What should `UpdateSettingCommand` look like?
Something like
```
export class UpdateSettingCommand {
	receiveTopRatedNotifications: boolean;
        (...)
}
``` ?
Or something closer  to what it is now but with column name validation?

Reply:
Maybe something like this https://github.com/mbrizic/ratethegame/pull/28/commits/e60f605ec000640f892482bd9b69e81edbde8ab8 ?
Now, true value inside `UpdateSettingCommand` indicates that we want to change the specific setting

Reply:
Additional column name validation added https://github.com/mbrizic/ratethegame/pull/28/commits/f9684f74f141d1e4238e38f514b81f34a6cf1e1f